sbrnet_core/sbrnet/dataset.py

```diff
@@ -67,6 +67,19 @@ def __getitem__(
         return stack, rfv, gt
 
 
+class ZarrData:
+    def __init__(self, directory: str, pattern: str):
+        self.directory = directory
+        self.pattern = pattern
+        self.open_zarrs = []
+    
+    # NOTE: ensure cache is larger than number of items
+    @cache
+    def __getitem__(self, index: int):
+        path = os.path.join(self.directory, self.pattern.format(index))
+        with TiffFile(path) as img:
+            return zarr.open(img.aszarr())
```

Comment:
how do we ensure this?

Reply:
I just checked the documentation and the cache is unbounded so it'll grow infinitely. So no modification needed.

Reply:
ok cool, youre changes look good to me. im ready to merge if you are